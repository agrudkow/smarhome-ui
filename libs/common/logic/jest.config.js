module.exports = {
  name: 'common-logic',
  preset: '../../../jest.config.js',
  transform: {
    '^.+\\.[tj]sx?$': 'ts-jest',
  },
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx', 'html'],
  coverageDirectory: '../../../coverage/libs/common/logic',
};
