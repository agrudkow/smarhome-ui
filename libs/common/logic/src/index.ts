export * from './lib/use-accordion';
export * from './lib/use-window-dimensions';
