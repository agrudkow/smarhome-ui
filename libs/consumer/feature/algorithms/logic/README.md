# consumer-feature-algorithms-logic

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `nx test consumer-feature-algorithms-logic` to execute the unit tests via [Jest](https://jestjs.io).
