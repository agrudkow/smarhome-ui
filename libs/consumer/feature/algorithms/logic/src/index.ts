export * from './lib/algorithm.interface';
export * from './lib/algorithms-cells-parser';
export * from './lib/algorithms-data-parser';
