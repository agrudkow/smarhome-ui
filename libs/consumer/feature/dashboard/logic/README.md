# consumer-feature-dashboard-logic

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `nx test consumer-feature-dashboard-logic` to execute the unit tests via [Jest](https://jestjs.io).
