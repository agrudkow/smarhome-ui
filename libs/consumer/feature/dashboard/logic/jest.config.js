module.exports = {
  name: 'consumer-feature-dashboard-logic',
  preset: '../../../../../jest.config.js',
  transform: {
    '^.+\\.[tj]sx?$': 'ts-jest',
  },
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx', 'html'],
  coverageDirectory:
    '../../../../../coverage/libs/consumer/feature/dashboard/logic',
};
