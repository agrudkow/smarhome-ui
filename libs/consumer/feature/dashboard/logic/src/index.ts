export * from './lib/consumer-feature-dashboard-logic';
