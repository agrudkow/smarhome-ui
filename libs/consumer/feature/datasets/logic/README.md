# consumer-feature-datasets-logic

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `nx test consumer-feature-datasets-logic` to execute the unit tests via [Jest](https://jestjs.io).
