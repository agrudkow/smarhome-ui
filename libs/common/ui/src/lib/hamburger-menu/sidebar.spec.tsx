import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { customRender } from '@smarthome/common/logic';
import Sidebar from './sidebar';
import { fireEvent } from '@testing-library/react';

const itemsList = [
  { icon: <div />, route: '', text: '' },
  { icon: <div />, route: '', text: '' },
  { icon: <div />, route: '', text: '' },
];

describe(' Sidebar', () => {
  it('should render successfully', () => {
    const { baseElement } = customRender(
      <Router>
        <Sidebar show={true} items={itemsList} onShowSidebarClick={jest.fn()} />
      </Router>
    );
    expect(baseElement).toBeTruthy();
  });

  it('should match snapshot', () => {
    const { baseElement } = customRender(
      <Router>
        <Sidebar show={true} items={itemsList} onShowSidebarClick={jest.fn()} />
      </Router>
    );
    expect(baseElement.firstChild).toMatchSnapshot();
  });

  it('should call handler function on a button click', () => {
    const handleClick = jest.fn();
    const { getByText } = customRender(
      <Router>
        <Sidebar
          show={true}
          items={itemsList}
          onShowSidebarClick={handleClick}
        />
      </Router>
    );
    fireEvent.click(getByText('Collapse sidebar'));
    expect(handleClick).toHaveBeenCalledTimes(1);
  });
});
