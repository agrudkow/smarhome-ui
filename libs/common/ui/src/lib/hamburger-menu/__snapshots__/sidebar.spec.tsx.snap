// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` Sidebar should match snapshot 1`] = `
.c3 {
  height: 45px;
  padding: 12px 16px;
  background-color: #4E4E4E;
  border-left: 2px solid #1BBDCE;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-transition: background-color 0.3s ease-in-out,border-left 0.3s ease-in-out;
  transition: background-color 0.3s ease-in-out,border-left 0.3s ease-in-out;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c3:hover {
  background-color: #4E4E4E;
}

.c5 {
  color: #555;
  font-size: 14px;
  line-height: 20px;
  -webkit-letter-spacing: -0.3px;
  -moz-letter-spacing: -0.3px;
  -ms-letter-spacing: -0.3px;
  letter-spacing: -0.3px;
  color: #1BBDCE;
  overflow: hidden;
  white-space: nowrap;
  padding-top: 2px;
  -webkit-transition: max-width 0.3s ease-in-out,opacity 0.3s ease-in-out, padding-left 0.3s ease-in-out,color 0.3s ease-in-out;
  transition: max-width 0.3s ease-in-out,opacity 0.3s ease-in-out, padding-left 0.3s ease-in-out,color 0.3s ease-in-out;
  max-width: 220px;
  opacity: 1;
  padding-left: 20px;
}

.c4 {
  min-width: 24px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c4 path {
  -webkit-transition: fill 0.3s ease-in-out;
  transition: fill 0.3s ease-in-out;
  fill: #1BBDCE;
}

.c2 {
  list-style-type: none;
  margin: 0;
  padding: 0;
}

.c1 {
  position: absolute;
  width: 220px;
  height: calc( 100vh - 40px );
  z-index: 9999;
  top: 40px;
  background-color: #2E2C2D;
  -webkit-transition: width 0.3s ease-in-out,left 0.3s ease-in-out;
  transition: width 0.3s ease-in-out,left 0.3s ease-in-out;
  color: #DFE3D5;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  overflow: hidden;
}

.c10 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  overflow: hidden;
  min-height: calc( 100vh - 40px );
}

.c6 {
  position: relative;
  left: 0;
  bottom: 0;
  height: 40px;
  border-top: 1px solid black;
  padding: 10px 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

.c8 {
  color: #555;
  font-size: 14px;
  line-height: 20px;
  -webkit-letter-spacing: -0.3px;
  -moz-letter-spacing: -0.3px;
  -ms-letter-spacing: -0.3px;
  letter-spacing: -0.3px;
  color: #DFE3D5;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  overflow: hidden;
  white-space: nowrap;
  -webkit-transition: max-width 0.3s ease-in-out,opacity 0.3s ease-in-out, padding 0.3s ease-in-out;
  transition: max-width 0.3s ease-in-out,opacity 0.3s ease-in-out, padding 0.3s ease-in-out;
  max-width: 220px;
  opacity: 1;
  padding: 2px 10px 0;
}

.c7 {
  max-width: 24px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-transition: -webkit-transform 0.3s ease-in-out;
  -webkit-transition: transform 0.3s ease-in-out;
  transition: transform 0.3s ease-in-out;
  -webkit-transform-origin: center;
  -ms-transform-origin: center;
  transform-origin: center;
  -webkit-transform: rotateZ(180deg);
  -ms-transform: rotateZ(180deg);
  transform: rotateZ(180deg);
}

.c9 {
  display: block;
  position: fixed;
  background-color: rgba(0,0,0,0.3);
  height: 100%;
  width: 100%;
  z-index: 300;
  height: calc( 100vh - 40px );
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

@media (min-width:769px) {
  .c5 {
    font-size: 14px;
    line-height: 20px;
  }
}

@media (min-width:1024px) {
  .c5 {
    font-size: 16px;
    line-height: 24px;
  }
}

@media (min-width:1024px) {
  .c5 {
    font-size: 20px;
    line-height: 26px;
  }
}

@media (min-width:769px) {
  .c1 {
    position: -webkit-sticky;
    position: -webkit-sticky;
    position: sticky;
    -webkit-transition: flex-basis 0.3s ease-in-out,left 0.3s ease-in-out;
    transition: flex-basis 0.3s ease-in-out,left 0.3s ease-in-out;
    -webkit-flex-basis: 220px;
    -ms-flex-preferred-size: 220px;
    flex-basis: 220px;
  }
}

@media (min-width:1024px) {
  .c1 {
    z-index: 0;
  }
}

@media (min-width:769px) {
  .c8 {
    font-size: 14px;
    line-height: 20px;
  }
}

@media (min-width:1024px) {
  .c8 {
    font-size: 16px;
    line-height: 24px;
  }
}

@media (min-width:1024px) {
  .c8 {
    font-size: 20px;
    line-height: 26px;
  }
}

@media (min-width:1024px) {
  .c9 {
    display: none;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <ul
        class="c2"
      >
        <li>
          <a
            aria-current="page"
            class="c3 active"
            data-testid="styled-sidebar-list-item"
            href="/"
          >
            <div
              class="c4"
            >
              <div />
            </div>
            <div
              class="c5"
              data-testid="text"
            />
          </a>
        </li>
        <li>
          <a
            aria-current="page"
            class="c3 active"
            data-testid="styled-sidebar-list-item"
            href="/"
          >
            <div
              class="c4"
            >
              <div />
            </div>
            <div
              class="c5"
              data-testid="text"
            />
          </a>
        </li>
        <li>
          <a
            aria-current="page"
            class="c3 active"
            data-testid="styled-sidebar-list-item"
            href="/"
          >
            <div
              class="c4"
            >
              <div />
            </div>
            <div
              class="c5"
              data-testid="text"
            />
          </a>
        </li>
      </ul>
      <div
        class="c6"
      >
        <div
          class="c7"
        >
          <svg
            height="20px"
            version="1.1"
            viewBox="0 0 448 512"
            width="20px"
            xmlns="http://www.w3.org/2000/svg"
            xmlns:xlink="http://www.w3.org/1999/xlink"
          >
            <path
              d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34zm192-34l-136-136c-9.4-9.4-24.6-9.4-33.9 0l-22.6 22.6c-9.4 9.4-9.4 24.6 0 33.9l96.4 96.4-96.4 96.4c-9.4 9.4-9.4 24.6 0 33.9l22.6 22.6c9.4 9.4 24.6 9.4 33.9 0l136-136c9.4-9.2 9.4-24.4 0-33.8z"
              fill="#DFE3D5"
            />
          </svg>
        </div>
        <div
          class="c8"
        >
          Collapse sidebar
        </div>
      </div>
    </div>
    <div
      class="c9"
    />
    <div
      class="c10"
    />
  </div>
</div>
`;
